src/libAtomVM/nifs.c

```diff
@@ -3469,6 +3489,47 @@ static term nif_atomvm_close_avm_pack(Context *ctx, int argc, term argv[])
     return OK_ATOM;
 }
 
+// AtomVM extension
+static term nif_atomvm_get_entry_point(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+
+    term name = argv[0];
+    if (UNLIKELY(!term_is_atom(name))) {
+        RAISE_ERROR(BADARG_ATOM);
+    }
+
+    int name_atom_id = term_to_atom_index(name);
+
+    struct ListHead *open_avm_packs = synclist_wrlock(&ctx->global->avmpack_data);
+    struct ListHead *item;
+    LIST_FOR_EACH (item, open_avm_packs) {
+        struct AVMPackData *avmpack_data = GET_LIST_ENTRY(item, struct AVMPackData, avmpack_head);
+        if (avmpack_data->name_atom_id == name_atom_id) {
+            uint32_t size;
+            const void *beam;
+            const char *module_name;
+            if (!avmpack_find_section_by_flag(avmpack_data->data, BEAM_START_FLAG, &beam, &size, &module_name)) {
+                synclist_unlock(&ctx->global->avmpack_data);
+                return UNDEFINED_ATOM;
+            }
+
+            int module_name_len = strlen(module_name);
+            int needed = term_binary_heap_size(module_name_len);
+            if (UNLIKELY(memory_ensure_free(ctx, needed) != MEMORY_GC_OK)) {
+                synclist_unlock(&ctx->global->avmpack_data);
+                RAISE_ERROR(OUT_OF_MEMORY_ATOM);
+            }
+            term entry_bin = term_from_literal_binary(module_name, module_name_len, &ctx->heap, ctx->global);
+            synclist_unlock(&ctx->global->avmpack_data);
+            return entry_bin;
+        }
+    }
+    synclist_unlock(&ctx->global->avmpack_data);
+
+    return UNDEFINED_ATOM;
```

Comment:
Should I stick to the `error` | `{ok, result}` convention?

Reply:
`error` | `{ok, result}` or `{error, Reason}` | `{ok, result}` seem like a good convention to stick to.

Reply:
@bettio If it's possible to include the `Reason` on `error` as @UncleGrumpy mentioned would be awesome but not required.